helpers/Physics.h

```diff
@@ -0,0 +1,380 @@
+			angX = -Math_Atan2F(sqrtf(SQ(workVec.x) + SQ(workVec.z)), workVec.y);
+			pPrevRot->y = angY;
+			pPrevRot->x = angX;
+		}
+		
+		Matrix_RotateY(angY, MTXMODE_NEW);
+		Matrix_RotateX(angX, MTXMODE_APPLY);
+		Matrix_MultZ(ABS(param->lengthList[i]) * param->gfx.scale.z, &posAdd);
+		
+		// Pushes limbs away from selected Vec3f points
+		for (s32 i = 0; i < param->spheres.numVec; i++) {
+			Vec3f tempPosAdd = {
+				pPrevPos->x + posAdd.x,
+				pPrevPos->y + posAdd.y,
+				pPrevPos->z + posAdd.z
+			};
+			f32 distTo = Math_Vec3f_DistXYZ(&tempPosAdd, &param->spheres.vecList[i]);
+			
+			if (distTo < param->spheres.dist) {
+				s16 yaw = Math_Vec3f_Yaw(&param->spheres.vecList[i], &tempPosAdd);
+				s16 pitch = Math_Vec3f_Pitch(&param->spheres.vecList[i], &tempPosAdd);
+				posAdd.x += Math_SinS(headRotY + 0x8000) * (param->spheres.dist - distTo) * 0.5f;
+				posAdd.z += Math_CosS(headRotY + 0x8000) * (param->spheres.dist - distTo) * 0.5f;
+				// posAdd.y += Math_SinS((-RADF_TO_BINANG(angX))) * (param->spheres.dist - distTo) * 0.5;
+				velAdj.x += Math_SinS(headRotY + 0x8000) * (param->spheres.dist - distTo) * 0.5f;
+				velAdj.z += Math_CosS(headRotY + 0x8000) * (param->spheres.dist - distTo) * 0.5f;
+				
+				pPrevRot->y = angY =  Math_Atan2F(posAdd.z, posAdd.x);
+				pPrevRot->x = angX = -Math_Atan2F(sqrtf(SQ(posAdd.x) + SQ(posAdd.z)), posAdd.y);
+			}
+		}
+		
+		workVec.x = pPos->x;
+		workVec.y = pPos->y;
+		workVec.z = pPos->z;
+		
+		pPos->x = pPrevPos->x + posAdd.x;
+		pPos->y = pPrevPos->y + posAdd.y;
+		pPos->z = pPrevPos->z + posAdd.z;
+		
+		pVel->x = (pPos->x - workVec.x + velAdj.x) * param->velMult;
+		pVel->y = (pPos->y - workVec.y + velAdj.y) * param->velMult;
+		pVel->z = (pPos->z - workVec.z + velAdj.z) * param->velMult;
+		
+		vel[i] = (Vec3f) {
+			CLAMP(pVel->x, -param->maxVel, param->maxVel),
+			CLAMP(pVel->y, -param->maxVel, param->maxVel),
+			CLAMP(pVel->z, -param->maxVel, param->maxVel),
+		};
+	}
+	
+	if (param->gfx.noDraw == true) {
+		Matrix_Pull();
+		
+		return;
+	}
+	
+	Gfx* disp = *dispType;
+	Mtx* matrix = Graph_Alloc(gfxCtx, param->J * sizeof(Mtx));
+	s16 y = RADF_TO_BINANG(rot[0].y);
+	Vec3f* pRot = &rot[0];
+	
+	gDPPipeSync(disp++);
+	gSPSegment(disp++, param->gfx.segID, matrix);
+	pPos = &pos[0];
+	
+	for (i = 0; i < param->J - 1; i++, pPos++, pRot++) {
+		Matrix_Translate(pPos->x, pPos->y, pPos->z, MTXMODE_NEW);
+		if (param->constraint.drawYawLimit) {
+			Math_SmoothStepToS(&y, RADF_TO_BINANG(rot[i].y), 3, DEGF_TO_BINANG(param->constraint.drawYawLimit), 1);
+			Matrix_RotateY(BINANG_TO_RAD(y), MTXMODE_APPLY);
+		} else {
+			Matrix_RotateY(rot[i].y, MTXMODE_APPLY);
+		}
+		Matrix_RotateX(rot[i].x, MTXMODE_APPLY);
+		if (param->lengthList[i] < 0)
+			Matrix_Scale(param->gfx.scale.x, param->gfx.scale.y, -param->gfx.scale.z, MTXMODE_APPLY);
+		else
+			Matrix_Scale(param->gfx.scale.x, param->gfx.scale.y, param->gfx.scale.z, MTXMODE_APPLY);
+		Matrix_ToMtx(&matrix[i], 0, 0);
```

Comment:
The game may use the `file` argument if the floats are too big, passing 0 may crash the game then (though at that point the game was on its way to declaring itself crashed so idk if it is much of an issue)

At the very least pass an empty string instead
```suggestion
		Matrix_ToMtx(&matrix[i], "", 0);
```

But it may help if ever for debugging to pass a useful string
```suggestion
		Matrix_ToMtx(&matrix[i], "Physics_DrawDynamicStrand", 0);
```